storage/watchable_store.go

```diff
@@ -241,6 +242,59 @@ func (s *watchableStore) syncWatchingsLoop() {
 	}
 }
 
+// RangeAllUnsynced ranges on all unsynced watchings
+// and returns all key-value pairs and next revision.
+func (s *watchableStore) RangeAllUnsynced() ([]storagepb.KeyValue, int64, error) {
+	totalLimit := 0
+	minRev, maxRev := int64(math.MaxInt64), int64(math.MinInt64)
+	for w := range s.unsynced {
+		if w.cur > 0 && w.cur <= s.store.compactMainRev {
+			log.Printf("storage: %v", ErrCompacted)
+			delete(s.unsynced, w)
+			continue
+		}
+		if w.cur > s.store.currentRev.main {
+			log.Printf("storage: %v", ErrFutureRev)
+			delete(s.unsynced, w)
+			continue
+		}
+		totalLimit += cap(w.ch) - len(w.ch)
```

Comment:
It's legacy arg from `RangeHistory` but just confirmed that we can just pass 0 to get all.

https://github.com/coreos/etcd/blob/master/storage/backend/batch_tx.go#L88-L90

I will change. Thanks,